### ISSUE TITLE
displaying PDF files leads to black projector

### ISSUE THREAD

**Issue by donovaly** [OpenSlides/OpenSlides#3552]
- upload this PDF to the files of OS:
[test.pdf](https://github.com/OpenSlides/OpenSlides/files/1658353/test.pdf)
- display it

result: the projector preview window doesn't show the PDF but is completely black

Since in the projector it is displayed but fullscreen, try to make it smaller:

result: using the zoom button in the projector preview has no effect

Using the zoom buttons in the presentation control elements works and when clicking to zoom out the projector preview is now working

- now rotate the PDF via the presentation control elements

result: this works

- now zoom in via the presentation control elements

result: the projector preview gets black again, the projector itself is correct.

- now press the reset zoom button in the presentation control elements

result: now also the projector is black

**Comment by serge1peshcoff** [comment]
Can confirm, I've also faced it when testing my local installation. I've had some PDF files for which the black screen was displayed (although it was working OK for all images and other PDF files).

**Comment by emanuelschuetze** [comment]
I can't reproduce it with your test.pdf. After upload and project the file I see the pdf in the preview:
![issue3552](https://user-images.githubusercontent.com/2075427/35502575-b353eeb6-04dd-11e8-8ae1-130f9da979a0.png)

**Comment by serge1peshcoff** [comment]
Weird, I get the correct output if I'll open the projector window (picture 1) but the black screen on the preview (picture 2)

![image](https://user-images.githubusercontent.com/1060257/35510405-f984ee6e-0508-11e8-8952-66cc76afca87.png)
![image](https://user-images.githubusercontent.com/1060257/35510369-dc26798c-0508-11e8-86c0-f229a50659d6.png)

Also I'm getting the black screen for some of my PDFs (unfortunately I can't provide them as they contain personal info) but sometimes when I go to the next page it suddenly works. I'm getting no errors on the console, and the file is loaded properly (I've checked the Network tab of my devtools and everything is loaded perfectly, I don't have any 4xx or 5xx responses)

**Comment by serge1peshcoff** [comment]
It's most likely the browser issue. I've just tried to open it in Chrome (I've used Firefox before) and everything was working clearly.

@donovaly can you tell what browser are you using and if it'll work with another browser?

**Comment by FinnStutzenstein** [comment]
Yes, this might be a browser issue. The pdf is rendered in a canvas (so done by the browser), using [ng-pdf here](https://github.com/OpenSlides/OpenSlides/blob/master/openslides/mediafiles/static/templates/mediafiles/slide_mediafile.html#L3). Sometimes rotating helps to refresh the view. The question is: What can OpenSlides do to prevent this?

Maybe retriggering the rendering, but how do we detect this? Maybe someone has an idea..

**Comment by serge1peshcoff** [comment]
The latest version at the Github repo is `2.0.0` https://github.com/sayanee/angularjs-pdf, and the version stated in `bower.json` is `~1.3.0`. I suppose that probably the `angular-pdf` maintainers have already fixed it in the latest versions.
Can somebody try updating the version in `bower.json`, install and check it?

**Comment by serge1peshcoff** [comment]
I've tested it with Safari (version 11.0.3), Opera (version 48.0.2685.39), Firefox (version 58.0) and Chrome (version 63.0.3239.132), it doesn't work in Firefox so far.

Mac OS X 10.13.3 x64

**Comment by donovaly** [comment]
I user Firefox 58 (64bit).

> version stated in bower.json is ~1.3.0

I don't have a bower.json in OS 2.2 nor in Firefox's install folder. Where can I find it?

**Comment by jsaalfeld** [comment]
> Can somebody try updating the version in bower.json, install and check it?

I updated the Version in the bower.json to ~2.0.0, ran yarn and spun up my instance, but it still is black in FF 58.0.2

**Comment by jsaalfeld** [comment]
So i tested the displaying with a bunch of different sized files:

[test_010567.pdf](https://github.com/OpenSlides/OpenSlides/files/1731371/test_010567.pdf)
[test_022086.pdf](https://github.com/OpenSlides/OpenSlides/files/1731372/test_022086.pdf)
[test_027599.pdf](https://github.com/OpenSlides/OpenSlides/files/1731373/test_027599.pdf)
[test_033813.pdf](https://github.com/OpenSlides/OpenSlides/files/1731374/test_033813.pdf)
[test_040990.pdf](https://github.com/OpenSlides/OpenSlides/files/1731375/test_040990.pdf)
[test_048210.pdf](https://github.com/OpenSlides/OpenSlides/files/1731376/test_048210.pdf)
[test_055443.pdf](https://github.com/OpenSlides/OpenSlides/files/1731377/test_055443.pdf)
[test_069626.pdf](https://github.com/OpenSlides/OpenSlides/files/1731378/test_069626.pdf)
[test_074416.pdf](https://github.com/OpenSlides/OpenSlides/files/1731379/test_074416.pdf)
[test_081637.pdf](https://github.com/OpenSlides/OpenSlides/files/1731380/test_081637.pdf)
[test_088913.pdf](https://github.com/OpenSlides/OpenSlides/files/1731381/test_088913.pdf)
[test_095964.pdf](https://github.com/OpenSlides/OpenSlides/files/1731382/test_095964.pdf)
[test_102597.pdf](https://github.com/OpenSlides/OpenSlides/files/1731383/test_102597.pdf)
[test_108826.pdf](https://github.com/OpenSlides/OpenSlides/files/1731384/test_108826.pdf)
[test_119331.pdf](https://github.com/OpenSlides/OpenSlides/files/1731385/test_119331.pdf)
[test_120930.pdf](https://github.com/OpenSlides/OpenSlides/files/1731386/test_120930.pdf)
[test_122845.pdf](https://github.com/OpenSlides/OpenSlides/files/1731387/test_122845.pdf)
[test_125942.pdf](https://github.com/OpenSlides/OpenSlides/files/1731388/test_125942.pdf)
[test_127868.pdf](https://github.com/OpenSlides/OpenSlides/files/1731389/test_127868.pdf)
[test_130534.pdf](https://github.com/OpenSlides/OpenSlides/files/1731390/test_130534.pdf)
[test_131702.pdf](https://github.com/OpenSlides/OpenSlides/files/1731391/test_131702.pdf)
[test_132771.pdf](https://github.com/OpenSlides/OpenSlides/files/1731392/test_132771.pdf)
[test_139621.pdf](https://github.com/OpenSlides/OpenSlides/files/1731393/test_139621.pdf)
[test_162096.pdf](https://github.com/OpenSlides/OpenSlides/files/1731394/test_162096.pdf)
[test_171122.pdf](https://github.com/OpenSlides/OpenSlides/files/1731395/test_171122.pdf)
[test_183790.pdf](https://github.com/OpenSlides/OpenSlides/files/1731396/test_183790.pdf)

(the numbers behind the files are the size in bytes).

It blinds my eyes to see, that files that are 127kb and below are displayed without any issue in FF, and files of 128kb and above causing the projector to black out. This fine power-of-two-number seems to be to oddly specific to be random. Can one verify that it happens not just on my system (maybe 32 bit systems as well?)

**Comment by emanuelschuetze** [comment]
I can confirm it with Firefox 58.0.2 (64bit) on Linux (test_130534.pdf works fine, test_131702.pdf blacks out). I have no idea what could be the reason...

**Comment by jsaalfeld** [comment]
Sure. I will write a small Dockerfile to check out specific commits and do a binary search thorugh the commits from 2.1.1 (bbbad71e2df2615ac33744cf69ee7ad1dfcfd108) until the commits from 28 days ago (the opening date of this issue - b5157cd418577f6d12422d150d71350d4b935e28).

**Comment by jsaalfeld** [comment]
Dockerfile I used: https://github.com/jsaalfeld/OpenSlides-buildCheckout

Commits I tried (in that order):
* f0d840148e0ba016c07da1e275aaab0653a989de - doesn't work
* 074559ba23cb7627697024bb78519f42f4dc4eff - doesn't work
* 4963bfa7bffeb0521e15217b50023ed3a24c1956  - doesn't work
* 8c4ec1c218ba63675edd11d6ce79d8729a19d64f - doesn't work
* 65b51f89756d699cc1edc2022c7f09c897d87c40 - doesn't work
* 757bbc62e542dba5f0a98f6f7ebdddb7c8d9909e - doesn't work
* d758159bffacccf32991c8e6982bdd3fade49e76 - doesn't work

Since d758159bffacccf32991c8e6982bdd3fade49e76 is older than the bbbad71e2df2615ac33744cf69ee7ad1dfcfd108 commit. It can be boiled down to: It's not the code.

So what else could provoke this behaviour? Since it works on the OpenSlides Test-Instance, my take would be, that it is about the architecture. In particular: the Redis Server or Postges Server, since all tests have been made with the "simple" setup and this is the only variable in the mix.

**Comment by emanuelschuetze** [comment]
demo.openslides.org (with v2.1.1) works. Here the installed requirements:
```
$ pip freeze
Django==1.9.3
argparse==1.2.1
beautifulsoup4==4.4.1
cffi==0.8.6
chardet==2.3.0
colorama==0.3.2
cryptography==0.6.1
djangorestframework==3.3.2
html5lib==0.999
jsonfield==1.0.3
natsort==4.0.4
ndg-httpsclient==0.3.2
ply==3.4
pyOpenSSL==0.14
pyasn1==0.1.7
pycparser==2.10
pytz==2012c
requests==2.4.3
six==1.8.0
sqlparse==0.1.13
urllib3==1.9.1
wheel==0.24.0
wsgiref==0.1.2


$ ./node_modules/.bin/bower list
bower check-new     Checking for new versions of the project dependencies...
openslides /home/openslides/openslides-git
├── angular#1.5.8 (1.5.11 available, latest is 1.6.9)
├─┬ angular-animate#1.5.11 (latest is 1.6.9)
│ └── angular#1.5.8 incompatible with 1.5.11 (1.5.11 available, latest is 1.6.9)
├─┬ angular-bootstrap#2.1.4 (latest is 2.5.0)
│ └── angular#1.5.8 (1.6.9 available)
├── angular-bootstrap-colorpicker#3.0.25 (3.0.32 available)
├─┬ angular-chosen-localytics#1.5.1 (latest is 1.8.0)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├── chosen#1.7.0 (1.8.3 available)
│ └── jquery#3.1.1 incompatible with ^2.0.3 (2.2.4 available, latest is 3.3.1)
├── angular-ckeditor#1.0.3
├─┬ angular-file-saver#1.1.3
│ ├── blob-polyfill#1.0.20150320 (latest is 2.0.20171115)
│ └── file-saver.js#1.20150507.2
├─┬ angular-formly#8.4.1
│ ├── angular#1.5.8 (1.6.9 available)
│ └── api-check#7.5.5
├─┬ angular-formly-templates-bootstrap#6.2.0 (latest is 6.3.2)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├── angular-formly#8.4.1
│ ├── api-check#7.5.5
│ └─┬ bootstrap#3.3.7 (4.0.0 available)
│   └── jquery#3.1.1 (3.3.1 available)
├─┬ angular-gettext#2.3.9 (2.3.10 available)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ angular-messages#1.5.11 (latest is 1.6.9)
│ └── angular#1.5.8
├─┬ angular-pdf#1.3.1 (latest is 2.0.0)
│ ├── angular#1.5.8 (latest is 1.6.9)
│ └── pdfjs-dist#1.5.188 (latest is 2.0.385)
├─┬ angular-sanitize#1.5.11 (latest is 1.6.9)
│ └── angular#1.5.8
├── angular-scroll-glue#2.0.7 (latest is 2.2.0)
├─┬ angular-ui-router#0.3.2 (latest is 1.0.15)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ angular-ui-tree#2.22.5 (2.22.6 available)
│ └── angular#1.5.8 (1.5.11 available, latest is 1.6.9)
├─┬ angular-xeditable#0.5.0 (latest is 0.8.1)
│ └── angular#1.5.8 (1.6.9 available)
├── bootstrap-css-only#3.3.7 (latest is 4.0.0)
├─┬ bootstrap-ui-datetime-picker#2.4.3 (latest is 2.6.1)
│ └── angular-bootstrap#2.1.4 incompatible with ^1.2.0 (1.3.3 available, latest is 2.5.0)
├── ckeditor#4.6.1 (latest is 4.8.0)
├── docxtemplater#2.1.5 (latest is 3.3.1)
├── font-awesome-bower#4.5.0 (latest is 4.7.0)
├── jquery#3.1.1 (latest is 3.3.1)
├─┬ jquery.cookie#1.4.1
│ └── jquery#3.1.1 (3.3.1 available)
├── js-data#2.9.0 (latest is 3.0.2)
├─┬ js-data-angular#3.2.1
│ ├── angular#1.5.8 (1.6.9 available)
│ └── js-data#2.9.0 (2.10.1 available, latest is 3.0.2)
├── lodash#4.16.6 (latest is 4.17.5)
├─┬ ng-dialog#0.6.6 (latest is 1.4.0)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ ng-file-upload#11.2.3 (latest is 12.2.13)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ ngBootbox#0.1.4 (0.1.5 available, latest is 0.2.0)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├─┬ bootbox#4.4.0
│ │ └── bootstrap#3.3.7 (4.0.0 available)
│ ├── bootstrap#3.3.7 (4.0.0 available)
│ └── jquery#3.1.1 (3.3.1 available)
├─┬ ngstorage#0.3.11
│ └── angular#1.5.8 (1.6.9 available)
├── papaparse#4.1.4 (latest is 4.3.7)
├── pdfmake#0.1.27 (latest is 0.1.35)
└── roboto-fontface#0.6.1 (latest is 0.9.0)
```

**Comment by jsaalfeld** [comment]
Docker container build with the Dockerfile.old from the above mentioned github repository:

```
$ pip freeze
appdirs==1.4.3
asgi-redis==1.0.0
asgiref==1.1.2
async-timeout==2.0.0
attrs==17.4.0
autobahn==17.10.1
Automat==0.6.0
bleach==1.5.0
channels==1.0.3
constantly==15.1.0
daphne==1.4.2
Django==1.10.8
django-redis==4.7.0
django-redis-sessions==0.5.6
djangorestframework==3.4.7
html5lib==0.9999999
hyperlink==17.3.1
incremental==17.5.0
jsonfield==1.0.3
msgpack-python==0.5.4
packaging==16.8
psycopg2==2.6.2
pyparsing==2.2.0
PyPDF2==1.26.0
redis==2.10.6
roman==2.0.0
six==1.11.0
Twisted==17.9.0
txaio==2.8.2
zope.interface==4.4.3

$ ./node_modules/.bin/bower list
bower check-new     Checking for new versions of the project dependencies...
openslides /app
├── angular#1.5.8 (1.5.11 available, latest is 1.6.9)
├─┬ angular-animate#1.5.11 (latest is 1.6.9)
│ └── angular#1.5.8 incompatible with 1.5.11 (1.5.11 available, latest is 1.6.9)
├─┬ angular-bootstrap#2.1.4 (latest is 2.5.0)
│ └── angular#1.5.8 (1.6.9 available)
├── angular-bootstrap-colorpicker#3.0.32
├─┬ angular-chosen-localytics#1.5.1 (latest is 1.8.0)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├── chosen#1.8.3
│ └── jquery#3.1.1 incompatible with ^2.0.3 (2.2.4 available, latest is 3.3.1)
├── angular-ckeditor#1.0.3
├─┬ angular-file-saver#1.1.3
│ ├── blob-polyfill#1.0.20150320 (latest is 2.0.20171115)
│ └── file-saver.js#1.20150507.2
├─┬ angular-formly#8.4.1
│ ├── angular#1.5.8 (1.6.9 available)
│ └── api-check#7.5.5
├─┬ angular-formly-templates-bootstrap#6.2.0 (latest is 6.3.2)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├── angular-formly#8.4.1
│ ├── api-check#7.5.5
│ └── bootstrap#4.0.0
├─┬ angular-gettext#2.3.10
│ └── angular#1.5.8 (1.6.9 available)
├─┬ angular-messages#1.5.11 (latest is 1.6.9)
│ └── angular#1.5.8
├─┬ angular-pdf#1.3.1 (latest is 2.0.0)
│ ├── angular#1.5.8 (latest is 1.6.9)
│ └── pdfjs-dist#1.5.188 (latest is 2.0.385)
├─┬ angular-sanitize#1.5.11 (latest is 1.6.9)
│ └── angular#1.5.8
├── angular-scroll-glue#2.0.7 (latest is 2.2.0)
├─┬ angular-ui-router#0.3.2 (latest is 1.0.15)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ angular-ui-tree#2.22.6
│ └── angular#1.5.8 (1.5.11 available, latest is 1.6.9)
├─┬ angular-xeditable#0.5.0 (latest is 0.8.1)
│ └── angular#1.5.8 (1.6.9 available)
├── bootstrap-css-only#3.3.7 (latest is 4.0.0)
├─┬ bootstrap-ui-datetime-picker#2.4.3 (latest is 2.6.1)
│ └── angular-bootstrap#2.1.4 incompatible with ^1.2.0 (1.3.3 available, latest is 2.5.0)
├── ckeditor#4.6.1 (latest is 4.8.0)
├── docxtemplater#2.1.5 (latest is 3.3.1)
├── font-awesome-bower#4.5.0 (latest is 4.7.0)
├── jquery#3.1.1 (latest is 3.3.1)
├─┬ jquery.cookie#1.4.1
│ └── jquery#3.1.1 (3.3.1 available)
├── js-data#2.9.0 (latest is 3.0.2)
├─┬ js-data-angular#3.2.1
│ ├── angular#1.5.8 (1.6.9 available)
│ └── js-data#2.9.0 (2.10.1 available, latest is 3.0.2)
├── lodash#4.16.6 (latest is 4.17.5)
├─┬ ng-dialog#0.6.6 (latest is 1.4.0)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ ng-file-upload#11.2.3 (latest is 12.2.13)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ ngBootbox#0.1.5 (latest is 0.2.0)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├─┬ bootbox#4.4.0
│ │ └── bootstrap#4.0.0
│ ├── bootstrap#4.0.0
│ └── jquery#3.1.1 (3.3.1 available)
├─┬ ngstorage#0.3.11
│ └── angular#1.5.8 (1.6.9 available)
├── papaparse#4.1.4 (latest is 4.3.7)
├── pdfmake#0.1.27 (latest is 0.1.35)
└── roboto-fontface#0.6.1 (latest is 0.9.0)
```

This means the differences are:
```
% diff openslides_docker.txt openslides_demo.txt 
2,19c2,10
< appdirs==1.4.3
< asgi-redis==1.0.0
< asgiref==1.1.2
< async-timeout==2.0.0
< attrs==17.4.0
< autobahn==17.10.1
< Automat==0.6.0
< bleach==1.5.0
< channels==1.0.3
< constantly==15.1.0
< daphne==1.4.2
< Django==1.10.8
< django-redis==4.7.0
< django-redis-sessions==0.5.6
< djangorestframework==3.4.7
< html5lib==0.9999999
< hyperlink==17.3.1
< incremental==17.5.0
---
> Django==1.9.3
> argparse==1.2.1
> beautifulsoup4==4.4.1
> cffi==0.8.6
> chardet==2.3.0
> colorama==0.3.2
> cryptography==0.6.1
> djangorestframework==3.3.2
> html5lib==0.999
21,31c12,25
< msgpack-python==0.5.4
< packaging==16.8
< psycopg2==2.6.2
< pyparsing==2.2.0
< PyPDF2==1.26.0
< redis==2.10.6
< roman==2.0.0
< six==1.11.0
< Twisted==17.9.0
< txaio==2.8.2
< zope.interface==4.4.3
---
> natsort==4.0.4
> ndg-httpsclient==0.3.2
> ply==3.4
> pyOpenSSL==0.14
> pyasn1==0.1.7
> pycparser==2.10
> pytz==2012c
> requests==2.4.3
> six==1.8.0
> sqlparse==0.1.13
> urllib3==1.9.1
> wheel==0.24.0
> wsgiref==0.1.2
> 
41c35
< ├── angular-bootstrap-colorpicker#3.0.32
---
> ├── angular-bootstrap-colorpicker#3.0.25 (3.0.32 available)
44c38
< │ ├── chosen#1.8.3
---
> │ ├── chosen#1.7.0 (1.8.3 available)
57,58c51,53
< │ └── bootstrap#4.0.0
< ├─┬ angular-gettext#2.3.10
---
> │ └─┬ bootstrap#3.3.7 (4.0.0 available)
> │   └── jquery#3.1.1 (3.3.1 available)
> ├─┬ angular-gettext#2.3.9 (2.3.10 available)
70c65
< ├─┬ angular-ui-tree#2.22.6
---
> ├─┬ angular-ui-tree#2.22.5 (2.22.6 available)
92c87
< ├─┬ ngBootbox#0.1.5 (latest is 0.2.0)
---
> ├─┬ ngBootbox#0.1.4 (0.1.5 available, latest is 0.2.0)
95,96c90,91
< │ │ └── bootstrap#4.0.0
< │ ├── bootstrap#4.0.0
---
> │ │ └── bootstrap#3.3.7 (4.0.0 available)
> │ ├── bootstrap#3.3.7 (4.0.0 available)
```

is there something in the diffs that could explain that issue? Should i try setup a big environment to test if its something with the external redis/postgres server?

**Comment by jsaalfeld** [comment]
I came along an issue: Why is the Django Version 1.9.3 in the Demo instance, while the requirements state, it should be something above 1.10.4? And I'm sure there are other requirements vs. reality issues.

**Comment by jsaalfeld** [comment]
I got a running version without any PDF issues (https://github.com/jsaalfeld/OpenSlides-buildCheckout/blob/master/Dockerfile.demo-openslides) it has the following details: 
```
$ pip freeze
appdirs==1.4.3
asgi-redis==1.3.0
asgiref==1.1.2
async-timeout==2.0.0
attrs==17.4.0
autobahn==17.10.1
Automat==0.6.0
bleach==1.5.0
channels==1.0.3
constantly==15.1.0
daphne==1.4.2
Django==1.10.8
django-redis==4.7.0
django-redis-sessions==0.5.6
djangorestframework==3.4.7
html5lib==0.9999999
hyperlink==17.3.1
incremental==17.5.0
jsonfield==1.0.3
msgpack-python==0.5.5
packaging==16.8
psycopg2==2.6.2
pyparsing==2.2.0
PyPDF2==1.26.0
redis==2.10.6
roman==2.0.0
six==1.11.0
Twisted==17.9.0
txaio==2.8.2
zope.interface==4.4.3


$ ./node_modules/.bin/bower list
├── angular#1.5.8 (latest is 1.6.9)
├─┬ angular-animate#1.5.8 (latest is 1.6.9)
│ └── angular#1.5.8 (latest is 1.6.9)
├─┬ angular-bootstrap#2.1.3 (latest is 2.5.0)
│ └── angular#1.5.8 (1.6.9 available)
├── angular-bootstrap-colorpicker#3.0.25 (latest is 3.0.32)
├─┬ angular-chosen-localytics#1.5.0 (latest is 1.8.0)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├── chosen#1.8.3
│ └── jquery#3.1.0 incompatible with ^2.0.3 (2.2.4 available, latest is 3.3.1)
├── angular-ckeditor#1.0.3https://github.com/jsaalfeld/OpenSlides-buildCheckout/blob/master/Dockerfile.demo-openslides
├─┬ angular-file-saver#1.1.2 (latest is 1.1.3)
│ ├── blob-polyfill#1.0.20150320 (latest is 2.0.20171115)
│ └── file-saver.js#1.20150507.2
├─┬ angular-formly#8.4.0 (latest is 8.4.1)
│ ├── angular#1.5.8 (1.6.9 available)
│ └── api-check#7.5.5
├─┬ angular-formly-templates-bootstrap#6.2.0 (latest is 6.3.2)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├── angular-formly#8.4.0 (8.4.1 available)
│ ├── api-check#7.5.5
│ └── bootstrap#4.0.0
├─┬ angular-gettext#2.3.7 (latest is 2.3.10)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ angular-messages#1.5.8 (latest is 1.6.9)
│ └── angular#1.5.8
├─┬ angular-pdf#1.3.0 (latest is 2.0.0)
│ ├── angular#1.5.8 incompatible with 1.4.8 (1.4.8 available, latest is 1.6.9)
│ └── pdfjs-dist#1.3.100 (latest is 2.0.385)
├─┬ angular-sanitize#1.5.8 (latest is 1.6.9)
│ └── angular#1.5.8
├── angular-scroll-glue#2.0.7 (latest is 2.2.0)
├─┬ angular-ui-router#0.3.1 (latest is 1.0.15)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ angular-ui-tree#2.22.0 (latest is 2.22.6)
│ └── angular#1.5.8 (1.5.11 available, latest is 1.6.9)
├─┬ angular-xeditable#0.5.0 (latest is 0.8.1)
│ └── angular#1.5.8 (1.6.9 available)
├── bootstrap-css-only#3.3.6 (latest is 4.0.0)
├─┬ bootstrap-ui-datetime-picker#2.4.0 (latest is 2.6.1)
│ └── angular-bootstrap#2.1.3 incompatible with ^1.2.0 (1.3.3 available, latest is 2.5.0)
├── ckeditor#4.6.1 (latest is 4.8.0)
├── docxtemplater#2.1.5 (latest is 3.3.1)
├── font-awesome-bower#4.5.0 (latest is 4.7.0)
├── jquery#3.1.0 (latest is 3.3.1)
├─┬ jquery.cookie#1.4.1
│ └── jquery#3.1.0 (3.3.1 available)
├── js-data#2.9.0 (latest is 3.0.2)
├─┬ js-data-angular#3.2.1
│ ├── angular#1.5.8 (1.6.9 available)
│ └── js-data#2.9.0 (2.10.1 available, latest is 3.0.2)
├── lodash#4.16.0 (latest is 4.17.5)
├─┬ ng-dialog#0.6.4 (latest is 1.4.0)
│ └── angular#1.5.8 (1.5.11 available, latest is 1.6.9)
├─┬ ng-file-upload#11.2.3 (latest is 12.2.13)
│ └── angular#1.5.8 (1.6.9 available)
├─┬ ngBootbox#0.1.3 (latest is 0.2.0)
│ ├── angular#1.5.8 (1.6.9 available)
│ ├─┬ bootbox#4.4.0
│ │ └── bootstrap#4.0.0
│ ├── bootstrap#4.0.0
│ └── jquery#3.1.0 (3.3.1 available)
├─┬ ngstorage#0.3.11
│ └── angular#1.5.8 (1.6.9 available)
├── papaparse#4.1.2 (latest is 4.3.7)
├── pdfmake#0.1.27 (latest is 0.1.35)
└── roboto-fontface#0.6.0 (latest is 0.9.0)
``` 

The differences are still quiet siginificantly. Can you may look at suspicious packages that may broke it?

**Comment by emanuelschuetze** [comment]
I found the problematic package: pdfjs-dist!

It works with:
pdfjs-dist#1.3.100 (from angular-pdf#1.3.0)

It does NOT works with:
pdfjs-dist#1.5.188 (from angular-pdf#1.3.1)

@jsaalfeld Please make some more tests with latest versions of angular-pdf and pdfjs-dist and create a pull request of an useful combination.
